tangy-form-reducer.js

```diff
@@ -18,6 +18,14 @@ const tangyFormReducer = function (state = initialState, action) {
 
     case 'FORM_OPEN':
       newState = Object.assign({}, action.response) 
+      newState.items = action.itemsInDom.map(itemInDom => {
+        const matchingItemInResponse = newState.items.find(item => item.id === itemInDom.id)
+        if (matchingItemInResponse) {
+          return { ...itemInDom, ...matchingItemInResponse }
+        } else {
+          return itemInDom
+        }
+      })
```

Comment:
Note that the use of the spread operator `...` in objects is the same as `Object.assign({}, itemInDom, matchingItemInResponse)`.